<launch>
    <arg name="sensor_definition_file" default="single_cam_gui_config"/>
    <arg name="sensor_definition_file_path" default="$(find telecarla_gui)/config/$(arg sensor_definition_file).json"/>
    <arg name="role_name" default="ego_vehicle"/>
    <arg name="vehicle_status" default="/carla/$(arg role_name)/vehicle_status"/>
    <arg name="camera_topics" default="{
        front: /carla/$(arg role_name)/front/image,
        front_left: /carla/$(arg role_name)/front_left/image,
        front_right: /carla/$(arg role_name)/front_right/image,
        rear: /carla/$(arg role_name)/rear/image,
        rear_left: /carla/$(arg role_name)/rear_left/image,
        rear_right: /carla/$(arg role_name)/rear_right/image
    }"/>
    <arg name="wheel_config" default="$(find telecarla_gui)/config/wheel_config.yml"/>

    <group ns="gui">
        <node name="telecarla_gui" pkg="telecarla_gui" type="telecarla_gui_node" output="screen">
            <param name="sensor_definition_file_path" value="$(arg sensor_definition_file_path)"/>
            <param name="role_name" value="$(arg role_name)"/>
            <param name="vehicle_status" value="$(arg vehicle_status)"/>
            <rosparam param="camera" subst_value="True">$(arg camera_topics)</rosparam>
            <rosparam file="$(arg wheel_config)" command="load"/>
        </node>
    </group>
</launch>
